### ISSUE TITLE
Update coffea to be compatible with latest versions of awkward.

### ISSUE THREAD

**Issue by lgray** [scikit-hep/coffea#114]
There are failures in Initializer/Builder and the old striped package (maybe we just get rid of it anyway?)

If possible, fixes should be backwards compatible with older uproots.

**Comment by lgray** [comment]
The failure in builder is actually an awkward-array bug!

```python
from awkward import JaggedArray, Table

tab = Table(isloose=[True], x=[0.5])
jagged = JaggedArray.fromcounts([0,0,0,0,0,0,1,0,0,0], tab)
masked_out = jagged[jagged['isloose']]
```

fails like:
```
Traceback (most recent call last):
  File "fail.py", line 7, in <module>
    masked_out = jagged[jagged['isloose']]
  File "/anaconda3/envs/py36/lib/python3.6/site-packages/awkward/array/jagged.py", line 536, in __getitem__
    intheadsum = inthead.sum()
  File "/anaconda3/envs/py36/lib/python3.6/site-packages/awkward/array/base.py", line 165, in sum
    return self._reduce(self.numpy.add, 0, None, regularaxis)
  File "/anaconda3/envs/py36/lib/python3.6/site-packages/awkward/array/jagged.py", line 1447, in _reduce
    out[thyself.starts == thyself.stops] = identity
IndexError: boolean index did not match indexed array along dimension 0; dimension is 7 but corresponding boolean dimension is 10
```

**Comment by lgray** [comment]
Fixed in #116
